src/auspex/instruments/holzworth.py

```diff
@@ -73,23 +73,35 @@ def frequency(self):
         return float(v.split()[0])*1e-3
     @frequency.setter
     def frequency(self, value):
-        self.ch_query(":FREQ:{} GHz".format(value))
+        fmin = float((self.ch_query(":FREQ:MIN?")).split()[0])
```

Comment:
I guess so, each query takes 4 ms. What if we set `self.fmin`, `self.fmax`... upon connection?